src/data/actions/assets.js

```diff
@@ -121,46 +121,44 @@ export const uploadingAssets = count => ({
   count,
 });
 
-export const uploadAssetSuccess = (asset, response) => ({
+export const uploadAssetSuccess = response => ({
   type: assetActions.UPLOAD_ASSET_SUCCESS,
-  asset,
   response,
 });
 
-export const uploadAssetFailure = (file, response) => ({
+export const uploadAssetFailure = (asset, response) => ({
   type: assetActions.UPLOAD_ASSET_FAILURE,
-  file,
+  asset,
   response,
 });
 
 export const uploadAssets = (files, courseDetails) =>
   (dispatch, getState) => {
     dispatch(uploadingAssets(files.length));
-    files.forEach((file) => {
+    // gather all the promises into a single promise that can be returned
+    return Promise.all(files.map(file => (
       clientApi.postUploadAsset(courseDetails.id, file)
         .then((response) => {
           if (response.ok) {
-            response.json().then((data) => {
-              dispatch(uploadAssetSuccess(data.asset, response));
-              dispatch(getAssets(getState().request, courseDetails));
+            return response.json().then((json) => {
+              dispatch(uploadAssetSuccess(json));
+              // dispatch(getAssets(..)) returns a promise, so we return it
+              // up to chain it
+              return dispatch(getAssets(getState().request, courseDetails));
             });
-          } else {
-            dispatch(uploadAssetFailure(file, response));
           }
-        });
-    });
+          dispatch(uploadAssetFailure(file, response.status));
+          return undefined;
```

Comment:
I'm a little iffy on this. I was forced to add it because of [this](https://eslint.org/docs/rules/consistent-return) eslint rule: 
> Expected to return a value at the end of arrow function. (consistent return)

Does this seem okay to everyone?

Reply:
This is done the same way in the accessibility actions file: https://github.com/edx/studio-frontend/blob/master/src/data/actions/accessibility.js#L27

While I'm not 100% on it I think it's fine.